modules/imgcodecs/test/test_tiff.cpp

```diff
@@ -840,6 +840,81 @@ TEST(Imgcodecs_Tiff, readWrite_predictor)
     }
 }
 
+// See https://github.com/opencv/opencv/issues/23416
+TEST(Imgcodecs_Tiff, readWrite_all_types)
+{
+    int typesList[] = {
+        CV_8UC1,  CV_8UC3,  CV_8UC4,
+        CV_8SC1,  CV_8SC3,  CV_8SC4,
+        CV_16UC1, CV_16UC3, CV_16UC4,
+        CV_16SC1, CV_16SC3, CV_16SC4,
+        CV_32SC1, CV_32SC3, CV_32SC4,
+        CV_32FC1, CV_32FC3, CV_32FC4,
+        CV_64FC1, CV_64FC3, CV_64FC4
+    };
```

Comment:
It's better to implement it as parameterized test. In case of single option failure, it's more obvious. Also less side effects between iterations. See https://github.com/opencv/opencv/blob/4.x/modules/core/test/test_arithm.cpp#L2216 as example and documentation: http://google.github.io/googletest/reference/testing.html#TEST_P

Reply:
Thank you for your comment ! I'll commit to fix it.

Reply:
I think it will be OK. 

https://github.com/Kumataro/opencv/blob/8de13490979ae3f2e2485b123d2df0c1ea2229b6/modules/imgcodecs/test/test_tiff.cpp#L875-L881